Bottom: 0df212f6a91ff8be79f4cf2dc830b87cf94ea432
Top:    23132933b3dde6a91cc589fd1cefee461543ba85
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:56:01 +0000

Fix a few random places where we fail badly if the board outline polygon computes to NULL

Usually, the board outline polygon should default to being the world region, but if a
polygon computation fails, it may return NULL, and we can crash in a few places if that
happens.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index cd1b79e..56fa133 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1684,6 +1684,8 @@ fill_board_outline_holes (hidGC gc, const BoxType *drawn_area)
   struct outline_info info;
 
   ensure_board_outline ();
+  if (PCB->Data->outline == NULL)
+    return;
 
   memset (&polygon, 0, sizeof (polygon));
   polygon.Clipped = PCB->Data->outline;
